src/classes/HiFiCommunicator.ts

```diff
@@ -457,6 +461,13 @@ export class HiFiCommunicator {
         if (typeof (hiFiGain) === "number") {
             this._currentHiFiAudioAPIData.hiFiGain = Math.max(0, hiFiGain);
         }
+        if (typeof (userAttenuation) === "number") {
+            this._currentHiFiAudioAPIData.userAttenuation = userAttenuation;
+        }
+        if (typeof (userRolloff) === "number") {
+            // TODO: will this ever be negative legitimately?
```

Comment:
In my opinion, this style of type checking is not helpful to the user (as it mysteriously changes the user's data without warning,) doesn't really deliver any safety to either users or server, and makes the code harder to maintain, extend, and audit.

However, GIVEN that we are doing this anyway, AND that we have declared that (until we change our minds) we will not support negative values, then I think the answer to the question is that it will not legitimately be negative and it would be consistent with the existing code here to block negative values.